### ISSUE TITLE
Search on fields in DataTablesAjaxRequestTrait

### ISSUE THREAD

**Issue by thejoelinux** [allanmcarvalho/DataTables#28]
Hi Allan,

I think I have found a bug in the DataTablesAjaxRequestTrait; when using a filter for a specific column, around line 170 :

```php
            switch ($columnType) {
                case "integer":
                    if (is_numeric($params['search']['value'])) {
                        $where[] = [$paramColumn['name'] => $columnSearch];
                    }
                    break;
                case "decimal":
                    if (is_numeric($params['search']['value'])) {
                        $where[] = [$paramColumn['name'] => $columnSearch];
                    }
                    break;
                case 'string':
                    $where[] = ["{$paramColumn['name']} like" => "%$columnSearch%"];
                    break;
                default:
                    $where[] = ["{$paramColumn['name']} like" => "%$columnSearch%"];
                    break;
            }
```

The test `is_numeric($params['search']['value'])` (in case 'integer' and case 'decimal') is always false because $params is an array. What should be tested is `$paramColumn['search']['value']`.

It was strange, because I made a bug in the configuration, the column was falsely guessed as a string, and it somehow worked, as the condition 
```php
$where[] = ["{$paramColumn['name']} like" => "%$columnSearch%"];
```
was applied.

I don't know if i'm clear, but that's probably because I did not sleep the entire night to sort it out :smile: 

I'll fork and propose a PR ASAP.

Oh, and I think this is similar as the issue #3.

**Comment by thejoelinux** [comment]
I forgot to note that I use the 3.x branch; I'll check out the master.

**Comment by allanmcarvalho** [comment]
Good job. I merged your changes and created a pre release '1.6.1-rc'. Try, test and let me know if is everything ok. If yes, I will create the final release.

Now I am working in a new version of cakephp-datatables, more clean, with more coverage and easy, that will work on CakePHP >= 4.0. Possibly if not hard, I will try create a branch of it for CakePHP >= 3.7.

**Comment by thejoelinux** [comment]
Thank you very much, it works like a charm.

I saw the new branch and well, I have tried to migrate to CakePHP 4.0 this morning, but there is two dependencies (cakephp-datatables is one) that prevent me from doing this.

Have you seen I have made a little project (https://github.com/thejoelinux/cakephp-dt-test) as sample code for your plugin ? Maybe I could propose to contribute in your 4.0 branch better than working alone ?

Have a nice day.

**Comment by allanmcarvalho** [comment]
Nice! Later I will make release!

Sure, currently I am creating the base business rules and structure. But I think that one more pair of eyes can be very helpful. Maybe in next week, I can launch a alpha version and you can start to give feedback and some improvements. Do you have some app like WhatsApp? We can talk about it better.

Ps. I changed the project repo... is on my company repo now, because I believe that is better manager opensource projects in a organization repo... the name is wsssoftware/cakephp-datatables. In the future I will archive this repo.

**Comment by thejoelinux** [comment]
Noted for the new repository.

I have WhatsApp, yes. How do we make contact on it ? By phone number ? Tbh, I just begin to use it.

**Comment by allanmcarvalho** [comment]
Yes. By phone number... My english speaking isn't very good, but we can exchange some messages, files and others things.
Can you add I? After send me a message that I will save your phone.

+55 43991727288
